### PR TITLE
Tests: Add module_loader

### PR DESCRIPTION
**Pull request by SchoolGuy**
This should make sure that we never break the modules importer.

Issue #2544 was doubting that and these tests show that this shouldn't be the case.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
- We ignore this weird CentOS 8 Error for now.
- Codacy complains about old tech-debt that we won't solve now.

File: cobbler/api.py [hunk 1]
```diff
@@ -1401,15 +1402,15 @@ def status(self, mode, logger=None):
...
         Automatically import a directory tree full of distribution files.
 
         :param mirror_url: Can be a string that represents a path, a user@host syntax for SSH, or an rsync:// address.
                            If mirror_url is a filesystem path and mirroring is not desired, set network_root to
                            something like "nfs://path/to/mirror_url/root"
-        :param mirror_name:
+        :param mirror_name: The name of the mirror.
         :param network_root:
         :param autoinstall_file:
         :param rsync_flags:
```
Comment (nodeg) [inline cobbler/api.py:1416]: What about the other descriptions?

File: cobbler/module_loader.py [hunk 2]
```diff
@@ -116,22 +116,31 @@ def get_module_by_name(name):
...
     :param category: Field category in configuration file.
-    :type category: str
     :param field: Field in configuration file
-    :type field: str
     :param fallback_module_name: Default value used if category/field is not found in configuration file
-    :type fallback_module_name: str
-    :raises CX: if unable to find configuration file
-    :returns: module name
-    :rtype: str
+    :raises FileNotFoundError: If unable to find configuration file.
```
Comment (nodeg) [inline cobbler/module_loader.py:126]:
```suggestion
    :raises FileNotFoundError: If unable to find a configuration file.
```

File: cobbler/modules/managers/import_signatures.py [hunk 3]
```diff
@@ -29,6 +29,16 @@
 
 import magic
 
+from typing import List, Callable, Any
```
Comment (nodeg) [inline cobbler/modules/managers/import_signatures.py:32]:
```suggestion
from typing import List, Callable, Any

import magic
```
See Coday complain.

File: cobbler/modules/managers/import_signatures.py [hunk 4]
```diff
@@ -84,11 +87,15 @@ def import_walker(top, func, arg):
...
+    """
+    This class contains most of the magic behind the ``cobbler import`` command. It uses the signatures as a data source
+    and then automatically adds the detected distro to Cobbler.
+    """
 
-    def __init__(self, collection_mgr, logger):
+    def __init__(self, collection_mgr, logger: clogger.Logger):
         """
-        Constructor
+        Main constructor for out class.
```
Comment (nodeg) [inline cobbler/modules/managers/import_signatures.py:98]: What does "out class" exactly mean?

**Review by nodeg — APPROVED** [review]
I left some comments.

File: cobbler/modules/managers/import_signatures.py (hunk 4 above)
Comment (SchoolGuy) [reply cobbler/modules/managers/import_signatures.py:98]: Typo.  `r` and `t` are neighboring characters on my layout.

File: cobbler/api.py (hunk 1 above)
Comment (SchoolGuy) [reply cobbler/api.py:1416]: I havn't looked in detail into those. I will do a cleanup PR with deeper insights later on.

File: cobbler/modules/managers/import_signatures.py (hunk 3 above)
Comment (SchoolGuy) [reply cobbler/modules/managers/import_signatures.py]: Sorted now

**Pull request merged**